engine/gamesys/src/gamesys/components/comp_sound.cpp

```diff
@@ -406,6 +424,7 @@ namespace dmGameSystem
                 entry.m_Sound = sound;
                 entry.m_StopRequested = 0;
                 entry.m_PauseRequested = 0;
+                entry.m_isMessage = 1;
```

Comment:
Perhaps, instead of doing that slightly complex if-statement when the sound is done, we can do it here:

E.g.
```
entry.m_ShouldDispatchEvents = entry.m_PlayId != dmSound::INVALID_PLAY_ID;

if (callback != 0)
{
    entry.m_ShouldDispatchEvents = 1; //possibly check that the listener fragment is set as well
}
```


Then, the final check would become:
```
if (entry.m_ShouldDispatchEvent)
{
    DispatchSoundEvent();
}
```

Reply:
True, I should have looked at it one more time with fresh eyes.